File: components/common/src/androidMain/kotlin/com/mirego/pilot/components/ui/type/PilotKeyboardAutoCapitalization.kt
package com.mirego.pilot.components.ui.type

import androidx.compose.ui.text.input.KeyboardCapitalization
import com.mirego.pilot.components.InternalPilotComponentsApi
import com.mirego.pilot.components.type.PilotKeyboardAutoCapitalization

@InternalPilotComponentsApi
public val PilotKeyboardAutoCapitalization.composeValue: KeyboardCapitalization
    get() = when (this) {
        PilotKeyboardAutoCapitalization.None -> KeyboardCapitalization.None
        PilotKeyboardAutoCapitalization.Sentences -> KeyboardCapitalization.Sentences
        PilotKeyboardAutoCapitalization.Words -> KeyboardCapitalization.Words
        PilotKeyboardAutoCapitalization.Characters -> KeyboardCapitalization.Characters
    }

File: components/ios/base/Types/PilotKeyboardAutoCapitalization.swift
import Shared
import SwiftUI

extension PilotKeyboardAutoCapitalization {
    public var uiTextAutocapitalizationType: UITextAutocapitalizationType {
        switch self {
        case .none:
            return .none
        case .characters:
            return .allCharacters
        case .sentences:
            return .sentences
        case .words:
            return .words
        }
    }
}
